core/libp2p/transport/tcp/tcp_connection.hpp

```diff
@@ -74,7 +74,7 @@ namespace libp2p::transport {
     bool isClosed() const override;
 
    private:
-    boost::asio::io_context &context_;
+    std::shared_ptr<boost::asio::io_context> context_;
```

Comment:
since TcpConnection is created by tcp_transport, I guess here it is fine to leave reference to context. Connections will not outlive transport.

Reply:
ok, refactored...